node/src/components/consensus/protocols/highway.rs

```diff
@@ -710,4 +710,29 @@ where
     fn instance_id(&self) -> &C::InstanceId {
         self.highway.instance_id()
     }
+
+    fn recreate_timers(&self) -> Vec<ProtocolOutcome<I, C>> {
+        let mut outcomes = vec![ProtocolOutcome::ScheduleTimer(
+            Timestamp::now(),
+            TIMER_ID_PURGE_VERTICES,
+        )];
+
+        if self.highway.is_active() {
+            outcomes.push(ProtocolOutcome::ScheduleTimer(
+                Timestamp::now(),
+                TIMER_ID_ACTIVE_VALIDATOR,
+            ));
+        }
+
+        outcomes.extend(
+            self.synchronizer
+                .timestamps_to_add_vertices()
+                .into_iter()
+                .map(|timestamp| {
+                    ProtocolOutcome::ScheduleTimer(timestamp, TIMER_ID_VERTEX_WITH_FUTURE_TIMESTAMP)
+                }),
```

Comment:
There shouldn't be any vertices in the synchronizer since we don't download the protocol state during joining, right?

Reply:
I think you're right - but I think it's still correct to do this. _If_ there are any vertices stored for later, we need to also recreate timers for them, or they will get stuck (maybe only for a while, but still).

Reply:
> If there are any vertices stored for later, we need to also recreate timers for them, or they will get stuck (maybe only for a while, but still).

but there can't be any vertices stored for later that transition from `joiner` to `validator` b/c we're ignoring them.

Reply:
I understand, but I see this as a detail this method shouldn't care about. This method is to ask the consensus instance for outcomes to create any timers it might be using. The fact that some of them might not be relevant at the point it is called is not a concern for this method itself.